lib/builder/context.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Burrito.Builder.Context do
+  use TypedStruct
+
+  alias Burrito.Builder.Target
+
+  @type erts_location :: nil | {:release | :local | :url | :unpacked, term()}
+
+  typedstruct do
+    field :target, Target.t(), enforce: true
+    field :erts_location, erts_location() , enforce: true
+    field :cross_build, boolean, enforce: true
+    field :mix_release, Mix.Release.t(), enforce: true
+    field :work_dir, String.t(), enforce: true
+    field :self_dir, String.t(), enforce: true
+    field :halt, boolean(), default: false
```

Comment:
Can we rename this to `halted` to match `Plug.Conn.t()` and how it's used within Phoenix?

Similarly, if you change [Burrito.Builder.run_phase/2](https://github.com/burrito-elixir/burrito/pull/30/files#diff-34487e839ba990b506825b06c35c1f578a81453e09dbb98254c3a32f358bd1b2R129) slightly, you'll be able to pull the `exit(1)` up to the top-level, and end up with a simpler control-flow that gives you an opportunity for any possible cleanup, while centralizing error handling at the boundary of the system:

```elixir
  def run(%Context{} = context, []) do
    context
  end

  def run(%Context{} = context, [mod | phases]) when is_atom(mod) do
    case mod.execute(context) do
      %Context{halted: true} = new_context ->
        new_context

      %Context{} = new_context ->
        run(new_context, phases)

      other ->
        raise "expected #{inspect(mod)} to return Context, got: #{inspect(other)}"
    end
  end
```

This is essentially how `Plug.run/3` is implemented.